frontend/src/Components/Header/UserStatus.js

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import app from "../Base/base";
+import { Avatar, Button, Col, Tooltip } from "antd";
+import { Link } from "react-router-dom";
+import { useContext } from "react";
+import "./index.css";
+
+import { AuthContext } from "../Auth/Auth";
+
+const UserStatus = (props) => {
+  const [color, setColor] = useState("#3f3f3f");
+  const [userid, setUserid] = useState(null);
+  const [user_letter, setUser_letter] = useState("");
+
+  const { currentUser } = useContext(AuthContext);
+
+  // Update userid on AuthContext change
+  useEffect(() => {
+    const rgb = Math.floor(Math.random() * 16777215);
+    const random_color = "#" + rgb.toString(16);
+    setColor(random_color);
+    if (currentUser == null) {
+      setUser_letter("");
+      setUserid(null);
+    } else {
+      setUserid(currentUser.uid);
+      setUser_letter(currentUser.uid[0]);
+    }
+  }, [currentUser]);
+
+  const avatar = (
+    <Tooltip title={userid}>
+      <Avatar
+        style={{
+          backgroundColor: color,
+          verticalAlign: "middle",
+          marginRight: 25,
+        }}
+        size="large"
+      >
+        {user_letter}
+      </Avatar>
+    </Tooltip>
+  );
+
+  const login_button = (
+    <Link to="/login">
+      <Button type="primary">Sign in</Button>
+    </Link>
+  );
+
+  const logout_button = (
+    <Link to="/logged-out">
+      <Button
+        type="primary"
+        onClick={() => {
```

Comment:
You can simplify this to 
```
onClick={()=> app.auth().signOut()}
```

Reply:
I'd like to keep the link so that we redirect the user to a different page, away from any content that relies on him being logged in. 
I think it also gives the user good feedback after logging out and I'd like to develop that page further in the future to add some information/links/components to it for more intuitive navigation within the app.